draft-ietf-quic-recovery.md

```diff
@@ -460,21 +463,34 @@ The PTO period is the amount of time that a sender ought to wait for an
 acknowledgement of a sent packet.  This time period includes the estimated
 network roundtrip-time (smoothed_rtt), the variance in the estimate (4*rttvar),
 and max_ack_delay, to account for the maximum time by which a receiver might
-delay sending an acknowledgement.
+delay sending an acknowledgement.  When the PTO is armed for Initial or
+Handshake packet number spaces, the max_ack_delay is 0.
 
 The PTO value MUST be set to at least kGranularity, to avoid the timer expiring
 immediately.
 
+A sender computes its PTO timer every time an ack-eliciting packet is sent.
+When ack-eliciting packets are in-flight in multiple packet number spaces,
+the timer MUST be set for the packet number space with the earliest timeout,
+except for ApplicationData, which MUST be ignored until the handshake
+completes (See Section 4.1.1 of {{QUIC-TLS}}).  Not arming the PTO for
+ApplicationData prioritizes completing the handshake and prevents the server
+from sending a 1-RTT PTO packet before it has the keys to process a 1-RTT
+packet.  When probe packets from two different packet number spaces are sent
+when the PTO fires and both spaces have in-flight packets, this simplifies to
+setting the PTO on the lowest active packet number space.
+
 When a PTO timer expires, the PTO period MUST be set to twice its current
 value. This exponential reduction in the sender's rate is important because
 the PTOs might be caused by loss of packets or acknowledgements due to severe
-congestion.  The life of a connection that is experiencing consecutive PTOs is
-limited by the endpoint's idle timeout.
+congestion.  Even when there are ack-eliciting packets in-flight in multiple
+packet number spaces, the exponential increase in probe timeout occurs across
+all spaces to prevent excess load on the network.  For example, a timeout in
+the Initial packet number space doubles the length of the timeout in the
+Handshake packet number space.
```

Comment:
If I understand correctly,
When Initial PTO is 10ms and Handshake PTO is 15ms:
  when Initial PTO expires, Initial PTO becomes 10ms and Handshake PTO becomes 30ms

Does it means Handshake PTO need to be re-armed at the point of Initial PTO expiration?
It may mean that Handshake PTO never be fired if Initial PTO keeps firing. There is other
text L544 saying that Initial and Handshake ack-eliciting packet should be sent at
Initial PTO expires. Then probably Handshake PTO need to be rearmed because we sent
Handshake ack-eliciting packet.
So for me it looks like now Initial PTO dominates Handshake PTO until Initial phase is complete.
Is my understanding correct?

Reply:
The Initial and Handshake timeouts should always be the same length, because neither uses the peer's max_ack_delay.  But they could be started from different points in time.  

As long as you send both at once, then the Initial PTO dominates.